[Rosa Kang] iP

ro4k9 opened the pull request:
# Artemis
> With organization comes empowerment [(source)](https://www.goodreads.com/quotes/813660-with-organization-comes-empowerment#:~:text=Quote%20by%20Lynda%20Peterson%3A%20%E2%80%9CWith%20organization%20comes%20empowerment.%E2%80%9D)

Organize your tasks like a pro with Artemis 😎 Artemis revolutionizes the way you organize and arrange tasks. It is:
- Text-based 
- Easy to learn 
- ~~$0.99 per month~~ FREE to use 

All you need to do is 🔽 
1. Download it. [(click me!)](https://github.com/ro4k9/ip/releases/tag/A-Jar)
2. Double-click to open.
3. Add your tasks.
4. Artemis will manage your tasks from this point.

**IT IS as simple as that 🤍**

Features:
- [x] Managing tasklist
- [x] Managing deadlines
- [x] Managing events
- [x] Managing todos
- [x] Finding tasks
- [x] Load previously added tasks
- [ ] Reminders (coming soon) 

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
   public static void main(String[] args) throws IOException {
        new Duke("data/duke.txt").run();
    }
```

Commit:
Add Gradle support

ro4k9 committed:
Added Greet, Echo, Exit

ro4k9 committed:
Added the ability to store task and updated the list

ro4k9 committed:
Added the ability to mark as done

ro4k9 committed:
Added todos, events, and deadline

ro4k9 committed:
Updated test Ui testing and error handlers

ro4k9 committed:
Added delete

ro4k9 committed:
Added class Display for displaying the text, Updated UI Testing and error handling

ro4k9 committed:
Added comments for classes and methods

ro4k9 committed:
Edited Documentation page

ro4k9 committed:
store tasks in local disk

ro4k9 committed:
implementing lvl 7

ro4k9 committed:
Implementation of lvl7

ro4k9 committed:
Implement lvl 8

ro4k9 committed:
implement UI

ro4k9 committed:
no message

ro4k9 committed:
Implemented Sotrage class

ro4k9 committed:
Implemented class Parser

ro4k9 committed:
Implemented A-MoreOOP

ro4k9 committed:
Implemented A-Packages

ro4k9 committed:
Implemented A-JUnit

ro4k9 committed:
Implemented A-Jar

ro4k9 committed:
Implemented A-JavaDoc.

ro4k9 committed:
Implemented A-CodingStandard.

ro4k9 committed:
no message

ro4k9 committed:
Revert "no message"

This reverts commit d4866404b0f0e96a0c7946e21e0075162030ae40.

ro4k9 committed:
Added testcase

ro4k9 committed:
Edited JavaDoc

ro4k9 committed:
no message

ro4k9 committed:
Implemented "find" in Duke, TaskList, and Ui

ro4k9 committed:
Added class Command

ro4k9 committed:
Added class Command and more JavaDoc comment

ro4k9 committed:
Remove the commented-out code

ro4k9 committed:
Merge branch 'add-gradle-support' into A-Gradle

ro4k9 committed:
Add config directory and checkstyle configuration

ro4k9 committed:
Add whitespace and re-order import according to coding standards in java

MontyPython28 commented on docs/README.md:
```diff
@@ -1,29 +1,232 @@
-# User Guide
+# Duke Chatbot
+Duke chatbot is a java program that helps you to manage your tasks
```
I really like how you've added to the documentation. It sort of acts as a user guide, which is so useful. It's definitely easy to understand the product with this!

MontyPython28 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,253 @@
+package duke;
```
Should the class have a header comment?

MontyPython28 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,54 @@
+import java.util.Scanner;
+import java.io.IOException;
+
+/**
+ * Duke class provides the functionality for Duke chatbot.
+ *
+ * @author Rosa Kang
+ */
```
Maybe you can use the `/** header_comment_here*/` structure for the one-line header comments?

MontyPython28 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,57 @@
+/**
+ * Event class represents an event task.
+ *
+ * @author Rosa Kang
```
I like the author tags

MontyPython28 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,88 @@
+    }
+
+    /**
+     * Convert date in the format yyyy-MM-dd to MMM dd yyyy if the input is a valid date.
```
I think that for method header comments, first word should end with s (i.e. in this case, `Converts` instead of `Convert`)

MontyPython28 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    /**
+     * The method marks task at i as incomplete.
```
Maybe could just be `Marks task at i as incomplete.`

MontyPython28 reviewed (commented):
Overall, looking really good! It was very easy to read and understand. I especially appreciate the changes to the documentation.
I just have a few tiny suggestions- mainly about the header comments.

Vishwanth2210 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,253 @@
+
+    /**
+     * A method to display a list of task.
+     */
```
It's nice that your included comments for the methods to make it easier to understand. I feel that it would be better if you start them in the form `Displays ... `, `Marks ... ` and etc instead of putting `A method ... `.  

Additionally, since this method has an parameter, it would be good to include its description.

Vishwanth2210 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,253 @@
+
+            if (temp.length < 2) {
+                throw new EmptyDescriptionException("time for command event");
+            } else e = new Event(temp[0], Parser.convertDate(temp[1]));
```
I am not quite sure, but May I know the reason for this else statement, as I don't quite see any Egyptian brackets after this else statement?

Vishwanth2210 commented on docs/README.md:
````diff
@@ -1,29 +1,232 @@
+`list`
 
-## Usage
+Expected outcome:
+```
+  n____n
+ ( o v o )
+--u-----u---------------------------------
+-> Here are the tasks in your list:
+    1.[T][x] test1
+    2.[E][ ] test2  (at: Mon 2-4pm)
+    3.[D][ ] test5  (by: June 6th)
+```
+Description of the outcome.
+```
+Duke lists command in a chronogical order based on time you add the tasks.
+The first box of the task indicates the type of task (todo, event, deadline). 
+The second box indicates the completion. This is followed by the description of a task and date.
+```
````
Very nice and clean. Good summary, makes it easier to understand what to expect.

Vishwanth2210 commented on src/main/java/duke/Command.java:
```diff
@@ -0,0 +1,253 @@
+     * @param s               storage to deal with making changes to text file
+     * @param ui              display functionality of duke
+     */
+    public void addTodo(String taskDescription, TaskList tasks, Storage s, Ui ui) {
+        try {
+            if (taskDescription == null) {
+                throw new EmptyDescriptionException("todo");
+            }
+            Todo t;
+            t = new Todo(taskDescription);
+            tasks.addTask(t);
+            ui.tasks(t.toString(), tasks.getSize());
+            s.appendToFile(t.format());
+        } catch (EmptyDescriptionException e) {
+            System.out.println(e.getMessage());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
```
Good usage of spacing. Clean and doesn't break coding standards

Vishwanth2210 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+    /**
+     * format Deadline to be stored into txt file.
+     *
+     * @return String representation of Deadline for txt file
```
Good summary of the return statement. Makes it easier to understand the use and reason for the String being returned from this method.

Vishwanth2210 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,54 @@
+    protected Storage s;
+    protected TaskList tasks;
+
+    /**
+     * Constructor for Duke.
+     */
```
I feel that it would be better if there is a description of the parameter "path" here.

Vishwanth2210 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,55 @@
+/**
+ * Deadline class represents the task with a deadline.
+ *
+ * @author Rosa Kang
```
Good usage of author tags

Vishwanth2210 reviewed (commented):
Good usage of spacing and following coding standards. Just have a fix a few nits with the comments for methods.

ro4k9 committed:
Update build.gradle and delete spacing

ro4k9 committed:
Edit code according to the coding standards

ro4k9 committed:
Update build.gradle for implementation of javaFx

ro4k9 committed:
Set up JavaFx

ro4k9 committed:
Add JavaFX Controls

ro4k9 committed:
Create a new class DialogBox

ro4k9 committed:
Implement GUI

ro4k9 committed:
Add assertions

This is to test the correctness of the assumptions made in the program.

Add assertion to method mark, unmark, delete, find to verify if the index is not empty and beyond the bound of user tasklist.
Add assertion method to addTodo, addEvent, and addDeadline to verify that the task description is not empty, and if applicable the time is indicated.

Assertion will allow us to know there is an error in which the program deals with the user input. Using assertion is also a good coding practice.

ro4k9 committed:
Merge pull request #1 from ro4k9/branch-A-Assertions

A-Assertions

ro4k9 committed:
Delete dead code in class Duke

As GUI is implemented, codes for text-based UI and I/O redirection.

Remove main and run method.
Remove an old constructor for Duke.

The main and run method is no longer needed as class Launch is used to run the program.
As old constructor for Duke, which was used by the main method, is no longer in use.
Delete dead code is a good coding practice.

ro4k9 committed:
Improve coding quality

ro4k9 committed:
Merge pull request #2 from ro4k9/branch-A-CodeQuality

A-CodeQuality

ro4k9 committed:
Add gradle.yml file

ro4k9 committed:
Merge pull request #3 from ro4k9/branch-A-CI

A-CI

ro4k9 committed:
Change the attribute type 'by' in Deadline to LocalDate from String

ro4k9 committed:
Edit statements affected by a change in class Deadline.

ro4k9 committed:
Create InvalidDateException

ro4k9 committed:
Implement Reminders

ro4k9 committed:
Merge pull request #4 from ro4k9/branch-B-Reminders

BCD-Extension

ro4k9 committed:
Change the color of the background and button

ro4k9 committed:
Merge pull request #6 from ro4k9/branch-A-BetterGui

A-BetterGui

ro4k9 committed:
Move GUI related classes to a subpackage

ro4k9 committed:
Move storage related class to a subpackage

ro4k9 committed:
Move storage related classes to a subpackage storage

ro4k9 committed:
Improve upon coding standards

ro4k9 committed:
Rename to Artemis

ro4k9 committed:
Edit JavaDoc Comments

ro4k9 committed:
Edit JavaDocs comment

ro4k9 committed:
Add close function upon user input 'bye'

ro4k9 committed:
Update parseDate method

ro4k9 committed:
Improve coding standard for Junit test

ro4k9 committed:
Move Launcher and Main classes to duke package

ro4k9 committed:
Change name from DukeTest to ArtemisTest

ro4k9 committed:
Move Launcher and Main classes to duke package

ro4k9 committed:
Delete Main and Launcher classes in gui package

ro4k9 committed:
Update main class name

ro4k9 committed:
Update Artemis chatbot documentation

ro4k9 committed:
Set theme jekyll-theme-minimal

ro4k9 committed:
Set theme jekyll-theme-minimal

ro4k9 committed:
Edit Artemis documentation

ro4k9 committed:
Merge branch 'master' of https://github.com/ro4k9/ip

ro4k9 committed:
Edit Artemis Documentation

ro4k9 committed:
Change archiveBaseName for shadowJar to artemis
